### PR TITLE
Filtering args

### PR DESCRIPTION
**Pull request by dsirov**
Name-based and section-based filtering introduced.

### CODE REVIEW

**Comment by eikendev** [comment]
Hello, thank you for your submission! I'll have a look at it tomorrow :wink:

File: rplugin/python3/denite/source/man.py [hunk 1]
```diff
@@ -4,27 +4,39 @@
...
-        command = self.__cmd
+        command = ['man', self.SEARCH_TYPE_SWITCH[search_type]]
         command.append(word)
 
+        if sections:
+            command += ['--section', sections]
+
         process = subprocess.run(
             command,
-            check=True,
```
Comment (eikendev) [inline rplugin/python3/denite/source/man.py:27]: Good catch!

File: rplugin/python3/denite/source/man.py [hunk 2]
```diff
@@ -4,27 +4,39 @@
...
         self.kind = 'man'
 
-        self.__cmd = ['man', '-k']
-
     def gather_candidates(self, context):
         word = context['input']
 
         if len(word) == 0:
-            word = '.'
+            word = self._get_arg( context, 0, default='.' )
```
Comment (eikendev) [inline rplugin/python3/denite/source/man.py]: Please remove the whitespace around the parentheses.

File: rplugin/python3/denite/source/man.py [hunk 3]
```diff
@@ -4,27 +4,39 @@
...
+
+        sections = self._get_arg( context, 1, default=None )
```
Comment (eikendev) [inline rplugin/python3/denite/source/man.py]: Please remove the whitespace around the parentheses.

File: rplugin/python3/denite/source/man.py [hunk 4]
```diff
@@ -4,27 +4,39 @@
...
+
+        if word != '.':
+            search_type = self._get_arg( context, 2, default='desc' )
```
Comment (eikendev) [inline rplugin/python3/denite/source/man.py]: Please remove the whitespace around the parentheses.

File: rplugin/python3/denite/source/man.py [hunk 5]
```diff
@@ -36,3 +48,18 @@ def gather_candidates(self, context):
         candidates = [{'word': result, 'addr': result, 'kind': 'man'} for result in candidates]
 
         return candidates
+
+    @staticmethod
+    def _get_arg( context, idx, default ):
```
Comment (eikendev) [inline rplugin/python3/denite/source/man.py]: Please remove the whitespace around the parentheses.

**Review by eikendev — CHANGES_REQUESTED** [review]
I've left some comments. Overall, very nice, just some style adjustments.

File: rplugin/python3/denite/source/man.py (hunk 2 above)
Comment (dsirov) [reply rplugin/python3/denite/source/man.py]: Yep, my personal style spilled out here... :)
Fixed.

File: rplugin/python3/denite/source/man.py (hunk 3 above)
Comment (dsirov) [reply rplugin/python3/denite/source/man.py]: Fixed

File: rplugin/python3/denite/source/man.py (hunk 4 above)
Comment (dsirov) [reply rplugin/python3/denite/source/man.py]: Fixed

File: rplugin/python3/denite/source/man.py (hunk 5 above)
Comment (dsirov) [reply rplugin/python3/denite/source/man.py]: Fixed

**Pull request merged**